components/builder-scheduler/src/data_store.rs

```diff
@@ -262,6 +261,43 @@ impl DataStore {
                            r#"CREATE OR REPLACE FUNCTION set_project_state_ident_v1 (pid bigint, jid bigint, state text, ident text) RETURNS void AS $$
                                   UPDATE projects SET project_state=state, job_id=jid, project_ident=ident, updated_at=now() WHERE id=pid;
                            $$ LANGUAGE SQL VOLATILE"#)?;
+
+        // Change the primary key for the groups table
+        migrator.migrate(
+            "scheduler",
+            r#"ALTER TABLE IF EXISTS groups DROP CONSTRAINT IF EXISTS groups_pkey;"#,
+        )?;
+        migrator.migrate(
+            "scheduler",
+            r#"CREATE SEQUENCE IF NOT EXISTS groups_id_seq;"#,
+        )?;
+        migrator.migrate("scheduler", r#"ALTER TABLE IF EXISTS groups ALTER COLUMN id SET DEFAULT next_id_v1('groups_id_seq');"#)?;
+        migrator.migrate(
+            "scheduler",
+            r#"ALTER TABLE IF EXISTS groups ADD PRIMARY KEY (id);"#,
+        )?;
+        migrator.migrate("scheduler",
+                     r#"CREATE OR REPLACE FUNCTION insert_group_v2 (
+                                project_names text[],
+                                project_idents text[]
+                                ) RETURNS SETOF groups AS $$
+                                    DECLARE
+                                        g groups % rowtype;
+                                    BEGIN
+                                        INSERT INTO groups (group_state)
+                                        VALUES ('Pending') RETURNING * INTO g;
+                                        RETURN NEXT g;
+
+                                        FOR i IN array_lower(project_names, 1)..array_upper(project_names, 1)
+                                        LOOP
+                                            INSERT INTO projects (owner_id, project_name, project_ident, project_state)
+                                            VALUES
+                                                (g.id, project_names[i], project_idents[i], 'NotStarted');
+                                        END LOOP;
+                                        RETURN;
+                                    END
+                                $$ LANGUAGE plpgsql VOLATILE
+                                "#)?;
```

Comment:
You're clearly a bit ahead of me with regards to Postgres functions.  I just copied the existing `insert_group_v1` function and tweaked it a bit until it did what I wanted.  I'll look into CTEs more carefully.  You can see where `project_names` and `project_idents` come from by looking down on line 389 of this file.  It shows up in the diff.  I assume, based on the way they're generated, that they are the same length.